src/remoted/manager.c

```diff
@@ -372,8 +368,28 @@ void save_controlmsg(const keyentry * key, char *r_msg, size_t msg_length, int *
         default:
             break;
         }
-        if (strcmp(r_msg, HC_STARTUP) == 0) {
+        if (strncmp(r_msg, HC_STARTUP, strlen(HC_STARTUP)) == 0) {
             mdebug1("Agent %s sent HC_STARTUP from '%s'", key->name, aux_ip);
+
+            const char *json_err;
+            cJSON *agent_info = NULL;
+            cJSON *version = NULL;
+            if (agent_info = cJSON_ParseWithOpts(strchr(r_msg, '{'), &json_err, 0), agent_info) {
+                if (version = cJSON_GetObjectItem(agent_info, "version"), cJSON_IsString(version)) {
+                    if (compare_wazuh_versions(__ossec_version, version->valuestring) < 0) {
+                        /* Reply to the agent */
+                        char msg_err[OS_FLSIZE + 1] = "";
+                        snprintf(msg_err, OS_FLSIZE, "%s%s", CONTROL_HEADER, HC_INVALID_VERSION);
```

Comment:
I think we should create the JSON using cJSON and not just use a define.

Reply:
Done!